docs/source/contributing/documentation.rst

```diff
@@ -317,3 +317,75 @@ elements that are either a ``str`` or ``None``;
 ``Tuple[:class:`str`, :class:`int`]`` for a tuple of type
 ``(str, int)``; ``Tuple[:class:`int`, ...]`` for a tuple of variable
 length with only integers.
+
+
+Adding type hints to functions and parameters
+---------------------------------------------
+
+If you've never used type hints before, this is a good place to get started:
+https://realpython.com/python-type-checking/#hello-types.
+
+When adding type hints to manim, there are some guidelines that should be followed:
+
+* Coordinates have the typehint ``Sequence[float]``, e.g.
+
+.. code:: py
+
+    def set_points_as_corners(self, points: Sequence[float]) -> "VMobject":
+        """Given an array of points, set them as corner of the Vmobject."""
+* ``**kwargs`` has no typehint
+
+* Mobjects have the typehint "Mobject", e.g.
+
+.. code:: py
+
+    def match_color(self, mobject: "Mobject"):
+        """Match the color with the color of another :class:`~.Mobject`."""
+        return self.set_color(mobject.get_color())
+* Colors have the typehint ``Color``, e.g.
+
+.. code:: py
+
+    def set_color(self, color: Color = YELLOW_C, family: bool = True):
+        """Condition is function which takes in one arguments, (x, y, z)."""
+* As ``float`` and ``Union[int, float]`` are the same, use only ``float``
+
+* For numpy arrays use the typehint ``np.ndarray``
```

Comment:
numpy.typing also specifies "ArrayLike". Not sure it's what we want, just mentioning it here. 

https://numpy.org/devdocs/reference/typing.html

Numpy Type Hints in general are a work in progress as I understand.